### PR TITLE
add categories

### PR DESCRIPTION
**Pull request by joamatab**
- Separate components by categories folders for mypy to pass
- Separate docs into different categories

### CODE REVIEW

File: gdsfactory/components/__init__.py [hunk 1]
```diff
@@ -1,315 +1,439 @@
...
-
-from __future__ import annotations
-
-from gdsfactory.components.add_fiber_array_optical_south_electrical_north import (
-    add_fiber_array_optical_south_electrical_north,
+from gdsfactory.components import (
+    bends,
+    containers,
+    couplers,
+    die,
```
[the remark this answers is not in the document]
Comment (MatthewMckee4) [reply gdsfactory/components/__init__.py:5]: So will cause the same issue as before

[the remark this answers is not in the document]
Comment (MatthewMckee4) [reply gdsfactory/components/__init__.py:5]: perhaps could rename folder to "dies" (i think that is the right plural here)

[the remark this answers is not in the document]
Comment (tvt173) [reply gdsfactory/components/__init__.py:5]: As long as it's the plural noun and not the verb 😝

[the remark this answers is not in the document]
Comment (joamatab) [reply gdsfactory/components/__init__.py]: fixed, im a bit worried this PR may be too disruptive

[the remark this answers is not in the document]
Comment (MatthewMckee4) [reply gdsfactory/components/__init__.py]: What issues persist?

**Comment by MatthewMckee4** [comment]
Perhaps another addition we could make for consistency is only referencing components one way.

For example in several files, some components are accessed via:
- gf.c.component_name
- from gdsfactory.components import component_name
- from gdsfactory.components.component_group import component_name
- from gdsfactory.components.component_group.component_name import component_name

Personally, i think:
```python
from gdsfactory.components.component_group.component_name import component_name
```
is the best option.
```python
from gdsfactory.components.component_group import component_name
```
will cause mypy issues the same way as before,
```python
from gdsfactory.components import component_name
```
could cause circular imports, and the same for
```python
gf.c.component_name
```

File: gdsfactory/components/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (joamatab) [reply gdsfactory/components/__init__.py]: The tests were not passing, it should be fixed now

**Comment by joamatab** [comment]
This is ready to review

**Pull request merged**